x/claims/keeper/integration_test.go

```diff
@@ -0,0 +1,506 @@
+package keeper_test
+
+import (
+	"encoding/json"
+	"math"
+	"math/big"
+	"strconv"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/ethereum/go-ethereum/crypto"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/tx/signing"
+	govtypes "github.com/cosmos/cosmos-sdk/x/gov/types"
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	"github.com/tharsis/ethermint/crypto/ethsecp256k1"
+	"github.com/tharsis/ethermint/encoding"
+	"github.com/tharsis/ethermint/tests"
+	evmtypes "github.com/tharsis/ethermint/x/evm/types"
+	"github.com/tharsis/evmos/v3/app"
+	"github.com/tharsis/evmos/v3/testutil"
+	incentivestypes "github.com/tharsis/evmos/v3/x/incentives/types"
+	inflationtypes "github.com/tharsis/evmos/v3/x/inflation/types"
+
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
+	authsigning "github.com/cosmos/cosmos-sdk/x/auth/signing"
+	authtx "github.com/cosmos/cosmos-sdk/x/auth/tx"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+	abci "github.com/tendermint/tendermint/abci/types"
+	"github.com/tharsis/ethermint/server/config"
+	evm "github.com/tharsis/ethermint/x/evm/types"
+	"github.com/tharsis/evmos/v3/contracts"
+	"github.com/tharsis/evmos/v3/x/claims/types"
+)
+
+var _ = Describe("Claiming", Ordered, func() {
+	claimsAddr := s.app.AccountKeeper.GetModuleAddress(types.ModuleName)
+	distrAddr := s.app.AccountKeeper.GetModuleAddress(distrtypes.ModuleName)
+	stakeDenom := stakingtypes.DefaultParams().BondDenom
+	claimsDenom := types.DefaultClaimsDenom
+	accountCount := 4
+
+	actionValue := sdk.NewInt(int64(math.Pow10(5) * 10))
+	claimValue := actionValue.MulRaw(4)
+	totalClaimsAmount := sdk.NewCoin(claimsDenom, claimValue.MulRaw(int64(accountCount)))
+
+	// account initial balances
+	initClaimsAmount := sdk.NewInt(int64(math.Pow10(5) * 2))
+	initEvmAmount := sdk.NewInt(int64(math.Pow10(18) * 2))
+	initStakeAmount := sdk.NewInt(int64(math.Pow10(10) * 2))
+	delegateAmount := sdk.NewCoin(claimsDenom, sdk.NewInt(1))
+	initBalance := sdk.NewCoins(
+		sdk.NewCoin(stakeDenom, initStakeAmount),
+		sdk.NewCoin(claimsDenom, initClaimsAmount),
+		sdk.NewCoin(evmtypes.DefaultEVMDenom, initEvmAmount),
+	)
+
+	// account for creating the governance proposals
+	initClaimsAmount0 := sdk.NewInt(int64(math.Pow10(18) * 2))
+	initBalance0 := sdk.NewCoins(
+		sdk.NewCoin(stakeDenom, initStakeAmount),
+		sdk.NewCoin(claimsDenom, initClaimsAmount0),
+		sdk.NewCoin(evmtypes.DefaultEVMDenom, initEvmAmount),
```

Comment:
why so many denoms? I think you can just use `claimsDenom`

Reply:
This address:
- creates a new gov proposal - without `stake` coins, it was failing with `0stake is smaller than 100000000stake: insufficient funds`
- the gov proposal is an incentive proposal, which requires a smart contract deployment - without `aphoton`, this was failing with `failed to deduct full gas cost 76562500000000aphoton from the user`

Reply:
you can change the default denomination in the governance and the evm parameters

Reply:
If I use only one denomination, the claimed/unclaimed balances become harder to account for, because the same denomination will be used for evm gas and gov actions.  Should I still do the change?

Reply:
I think using a single denom is better to test the behavior we would expect on mainnet